lib/chef/azure/helpers/shared.rb

```diff
@@ -88,4 +89,36 @@ def report_status_to_azure (message, status_type)
       AzureExtensionStatus.log(@azure_status_file, message, status_type)
     end
   end
+
+  module DeleteNode
+    include ChefAzure::Shared
+
+    def delete_node
+      Chef::Log.info "Inside delete node call"
+      begin
+        Chef::Config.from_file("#{bootstrap_directory}/client.rb")
+
+        unless Chef::Config[:node_name]
+          ohai = Ohai::System.new
+          ohai.all_plugins
+          Chef::Config[:node_name] = ohai[:fqdn] || ohai[:machinename] || ohai[:hostname]

```

Comment:
Ok, so there's not a good way for us to reliably obtain the node name, and the scenario we think is not super common. In that case, I think the right thing is for us not to try to do this. Users can use knife-azure if they are interactively managing the VM -- the `--purge` option will clean up the node / client based on the user's knowledge of the node name.

We can revisit this once Chef has better support for obtaining claims directly from the identity.